packages/utils-compat-data/src/css.ts

```diff
@@ -50,71 +56,126 @@ const getTokens = (nodes: any[]): [string, string][] => {
+        if (featureName === tokenValue) {
+            return getUnsupportedBrowsers(featureSupport, tokenPrefix, browsers, tokenValue, context);
+        }
+    }
+
+    return null;
+};
+
+/**
+ * Determine if part of a CSS value is supported. Iterates through all
+ * sub-features in the provided context, using keys and `matches` data
+ * to test each tokenized string from the value.
+ */
+const getPartialValueUnsupported = (context: Identifier, value: ParsedValue, browsers: string[]): UnsupportedBrowsers | null => {
+    for (const [featureName, featureSupport] of Object.entries(context)) {
+        if (featureName === '__compat') {
+            continue;
+        }
+
+        const unsupported = getValueMatchesUnsupported(context, featureSupport, value, browsers) ||
+            getValueTokenUnsupported(context, featureName, featureSupport, value, browsers) ||
+            getPartialValueUnsupported(featureSupport, value, browsers);
+
+        if (unsupported) {
+            return unsupported;
+        }
+    }
+
+    return null;
+};
+
 /**
  * Determine if the provided CSS value is supported, first by looking for an
  * exact match for the full value, falling back to search for a partial match.
  */
-const getValueUnsupported = (context: Identifier, property: string, value: string, browsers: string[]): UnsupportedBrowsers | null => {
-    const [data, prefix, unprefixed] = getFeatureData(context, value);
-
-    if (data) {
-        return getUnsupportedBrowsers(data, prefix, browsers, unprefixed, data.__compat?.mdn_url ? undefined : context);
+const getValueUnsupported = (context: Identifier | undefined, property: string, value: string, browsers: string[]): UnsupportedBrowsers | null => {
+    /*
+     * Context is missing when a property was omitted due to full support (to reduce bundle size),
+     * but referenced CSS types with partial support may still exist (e.g. "color" and alpha_hex_value).
+     */
+    if (context) {
+        const [data, prefix, unprefixed] = getFeatureData(context, value);
+
+        if (data) {
+            return getUnsupportedBrowsers(data, prefix, browsers, unprefixed, data.__compat?.mdn_url ? undefined : context);
+        }
     }
 
-    if (property && types.has(property)) {
+    const parsedValue = parseValue(value);
+
+    // Check browser support for each CSS type associated with the property (if any).
+    if (types.has(property)) {
         for (const type of types.get(property)!) {
-            const result = getValueUnsupported(mdn.css.types[type], '', value, browsers);
+            const typeContext = mdn.css.types[type];
+            const result = typeContext && getPartialValueUnsupported(typeContext, parsedValue, browsers);
 
             if (result) {
                 return result;
             }
         }
     }
 
-    return getPartialValueUnsupported(context, value, browsers);
+    return context ? getPartialValueUnsupported(context, parsedValue, browsers) : null;
```

Comment:
Maybe I'm missing something, but why don't do a `return getPartialValueUnsupported(context, parsedValue, browsers)` inside the `if (context)`?

Reply:
Good question. I was originally maintaining code order for precedence between searching for partial value matches in this entry versus searching for partial value matches in referenced types. More generally this ties back to this API not being designed to report multiple issues (e.g. when two separate parts of the value have different unsupported browsers we just report the first).

Looking at this again I don't think that order is particularly important so I'm going to make the change you suggest. It would only matter in the rare case when both paths had unsupported browsers - either way we'll display something to the user and once they fix one issue, we'll report the next.